lang/cs/Org.Apache.REEF.Network/Group/Config/GroupCommConfigurationOptions.cs

```diff
@@ -44,7 +44,7 @@ public class Timeout : Name<int>
         {
         }
 
-        [NamedParameter("Retry times", defaultValue: "5")]
+        [NamedParameter("Retry times", defaultValue: "500")]
         public class RetryCount : Name<int>
```

Comment:
Actually speaking this number should be some monotonic function of number
of nodes in the communication group. More the number of nodes, greater time
it takes for driver to get all evaluators running, greater time it takes to
set up connection due to latencies and greater the chance of hitting a slow
node.

Dhruv

On Wed, Jul 8, 2015 at 1:08 PM, Markus Weimer <notifications@github.com>
wrote:

> In
> lang/cs/Org.Apache.REEF.Network/Group/Config/GroupCommConfigurationOptions.cs
> <https://github.com/apache/incubator-reef/pull/278#discussion_r34192330>:
>
> > @@ -44,7 +44,7 @@ public class Timeout : Name<int>
> >          {
> >          }
> >
> > -        [NamedParameter("Retry times", defaultValue: "5")]
> > +        [NamedParameter("Retry times", defaultValue: "500")]
> >          public class RetryCount : Name<int>
>
> Just setting it to an arbitrary number that happens to work seems
> dangerous. We should at least add the above reasoning to the comments of
> the code to make sure we remember why we choose 500 as the magic number.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-reef/pull/278/files#r34192330>.
>